src/components/carousel/index.jsx

```diff
@@ -0,0 +1,70 @@
+import PropTypes from "prop-types";
+import Slider from "react-slick";
+import { useEffect, useRef } from "react";
+
+const COMPONENT_CLASS_NAME = "c-carousel";
+
+const Carousel = ({ children, className, sliderProps }) => {
+	const sliderRef = useRef();
+	const { onReInit: sliderReInit } = sliderProps;
+
+	/**
+	 * This useEffect will enhance a11y in the following:
+	 * 1) It will add the role and aria-label to the react-slick component
+	 * 2) It will add needed roles and labels to each slide.
+	 * 3) If paging is being used it will set the current pager container to aria-current=true
+	 */
+	useEffect(() => {
+		const slider = sliderRef.current.querySelector(".slick-slider");
+		slider?.setAttribute("role", "region");
+		slider?.setAttribute("aria-label", "carousel");
+
+		const slides = slider?.querySelectorAll(".slick-slide");
+		slides?.forEach((slide) => {
+			slide.setAttribute("role", "group");
+			slide.setAttribute("aria-label", `slide ${parseInt(slide.dataset.index, 10) + 1}`);
+		});
+
+		const pager = slider?.querySelector(".slick-dots .slick-active > *");
+		if (pager) pager.setAttribute("aria-current", "true");
+	}, []);
+
+	/**
+	 * The onReInit is a callback of the react-slider.
+	 * It is being used to update the aria-current attribute on the pager if its being used.
+	 * It also checks to see if the caller passed in their own onReInit and if so executes it.
+	 */
+	const onReInit = () => {
+		const currentActive = sliderRef.current.querySelector(
+			'.slick-slider .slick-dots [aria-current="true"]'
+		);
+		const pager = sliderRef.current.querySelector(".slick-slider .slick-dots .slick-active > *");
+		if (currentActive && pager) {
+			currentActive.removeAttribute("aria-current");
+			pager.setAttribute("aria-current", "true");
+		}
```

Comment:
Should we perhaps clear any `aria-current` value instead of the first one it finds in case something interferes or there's an issue during the last update?

```suggestion
		const activeElements = sliderRef.current.querySelectorAll(
			'.slick-slider .slick-dots [aria-current]'
		);
		const pager = sliderRef.current.querySelector(".slick-slider .slick-dots .slick-active > *");
		if (activeElements.length && pager) {
			Array.from(activeElements).forEach((e) => e.removeAttribute("aria-current"));
			pager.setAttribute("aria-current", "true");
		}
```